academy/task.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+import asyncio
+import logging
+from collections.abc import Coroutine
+from typing import Any
+
+from academy.exception import SafeTaskExitError
+from academy.logging import execute_and_log_traceback
+
+logger = logging.getLogger(__name__)
+
+
+def exit_on_error(task: asyncio.Task[Any]) -> None:
+    """Task callback that raises SystemExit on task exception."""
+    if (
+        not task.cancelled()
+        and task.exception() is not None
+        and not isinstance(task.exception(), SafeTaskExitError)
+    ):
+        logger.error(
+            f'Exception in background task (name="{task.get_name()}"): '
+            f'{task.exception()!r}',
+        )
+        raise SystemExit(1)
+
+
+def spawn_guarded_background_task(
+    coro: Coroutine[Any, Any, Any],
+    *,
+    name: str,
+) -> asyncio.Task[Any]:
+    """Run a coroutine safely in the background.
+
+    Launches the coroutine as an asyncio task and sets the done
+    callback to [`exit_on_error()`][proxystore.utils.tasks.exit_on_error].
```

Comment:
For simplicity, I'd suggest making renaming it `_exit_on_error` and just updating this docstring to say "sets the done callback to raise a [SystemExit][SystemExit] if the task raises an exception."